Feature/Optimize PageCounter Queries Release Two [ENG-598]

pattisdr opened the pull request:
❗️ This is Part Two of Page Counter Migration - pointing against Part One temporarily so changes are more visible.
[Part One of PageCounter Migration here
](https://github.com/CenterForOpenScience/osf.io/pull/8800
)

## Purpose

Part One of the Page Counter Migration adds new fields to the Page Counter migration, migrates the data, and adds some code to populate all fields.

Part Two of the PageCounter migration updates the code to use the new guid/file/action/version fields.

<!-- Describe the purpose of your changes -->

## Changes

- Updates code to query/create new PageCounters using the new guid/file/version/action fields
- Modifies queries on conference view, osfstorage view

## QA Notes
Double check download counts on file detail page, file revisions page, conference submissions page, and file detail in API v2. Double check downloading files increases file downloads as expected.
 
## Side effects

<!--Any possible side effects? -->


## Ticket

https://openscience.atlassian.net/browse/ENG-598

pattisdr commented on addons/osfstorage/utils.py:
```diff
@@ -16,7 +16,7 @@
 
 
-def update_analytics(node, file_id, version_idx, action='download'):
+def update_analytics(node, file, version_idx, action='download'):
```
Passing in the file instead of the file_id here. Also works with preprints.

pattisdr commented on osf/migrations/0146_pagecounter_id.py:
```diff
@@ -0,0 +1,21 @@
+        migrations.AlterField(
+            model_name='pagecounter',
+            name='_id',
+            field=models.CharField(db_index=True, default=osf.models.base.generate_object_id, max_length=50, unique=True),
```
Migration to use generate_object_id to make the _id field instead of building it like: 'action:node_id:file_id:version_idx'. I left existing _ids as is, but don't know if we should migrate them as well (are we currently exposing this anywhere?) Also if we wouldn't ever expose this, perhaps the _id field could be removed entirely at this point.

erinspace replied:
My gut says if we're making new `PageCounter`s with new `_id`, we should migrate all of em for consistency and to help future-selves from being confused. But you bring up a good point - I don't see a need really for an object_id as they're never being used like in an API detail endpoint or something like that - as long as we don't query on them anymore I suppose!

pattisdr replied:
Yeah, I'm not sure which way to go, I'll have to get Steve to weigh in.  If they're not being used or queried on, then it's a nice backup for the migration.  But I agree it could cause confusion.

pattisdr replied:
Discussed w/ Steve in-person, I'm going to revert this and keep constructing long id as we were previously (a mash-up of the four required fields), even though we don't need to keep querying this way. This will keep things consistent for a bit, while we make sure we don't need to revert back to querying on _id for some reason.  After some time has elapsed and we're sure this is stable, then we can make a migration to turn everything into long ids.  Thanks for looking out @erinspace!

erinspace commented on framework/analytics/__init__.py:
```diff
@@ -30,33 +28,15 @@ def get_total_activity_count(user_id):
+def update_counter(guid, file, version, action, node_info=None):
     """Update counters for page.
 
     :param str page: Colon-delimited page key in analytics collection
```
Could update docstrings here for more clarity!

erinspace commented on addons/osfstorage/views.py:
```diff
@@ -105,7 +105,7 @@ def osfstorage_get_revisions(file_node, payload, target, **kwargs):
 
     version_count = file_node.versions.count()
     # Don't worry. The only % at the end of the LIKE clause, the index is still used
```
Does this comment still apply? Can't quite tell!

pattisdr replied:
I don't think it's relevant anymore!

erinspace reviewed (commented):
Ditto from part 1 - lookin awesome. A few small things!

pattisdr commented on osf/models/analytics.py:
```diff
@@ -99,34 +96,24 @@ def clean_page(page):
+            # After we're sure this is stable, we can stop writing to the _id field, and query on
+            # resource/file/action/version
             try:
                 model_instance = cls.objects.get(_id=cleaned_page)
```
Temporarily still looking up the page_counter on _id.  We're now writing to five fields on creation - _id, resoure, file, action, and version.  

After this has been out awhile and it's stable, then we can remove the _id field, using a long id or similar instead, and migrating all the old _ids in the db.  After that, we can start querying on resource/file/action/version combo.

pattisdr commented:
This is going to need to be brought up-to-date at some point, but I'll wait until we get Brian's stuff merged in.

pattisdr committed:
Merge pull request #9042 from corbinSanders/feature/ampersand-display-fix

Fix ampersand displaying as "&amp;" in Public/Private dialog box [ENG-53]

pattisdr committed:
Merge pull request #8939 from pattisdr/feature/PLAT1328dataverse_in_dataverse

Feature/Fix Dataverse Error [PLAT-1328]

pattisdr committed:
Merge pull request #8995 from pattisdr/feature/add_contrib_modal

Disable Add button after clicking submitting changes [ENG-130]

Johnetordoff committed:
add testmon support (#8988)

pattisdr committed:
Merge pull request #9068 from adlius/ENG-586_limit_submission_url_visibility

[ENG-586] Add ShowIfChronosSubmitter conditional field

Commit:
Adding handling for child to parent linking

brianjgeiger committed:
Respond to CR

Commit:
Resolved merge conflicts

Commit:
Merge branch 'CenterForOpenScience-feature/fearow' into feature/admin-userview-preprint

Commit:
Fixing misunderstanding. Adding in contributor_or_group_member_to for the node serializer`

brianjgeiger committed:
Clean up, plus aggressive api versioning

pattisdr committed:
Merge branch 'feature/editable-registrations' of https://github.com/CenterForOpenScience/osf.io into feature/ENG557_editable_registration_logging

# Conflicts:
#	api_tests/registrations/views/test_registration_detail.py

pattisdr committed:
Merge branch 'develop' into feature/editable-registrations

# Conflicts:
#	api/registrations/serializers.py
#	api_tests/nodes/views/test_node_detail.py
#	api_tests/preprints/views/test_preprint_detail.py
#	osf/models/preprint.py

pattisdr committed:
Add dropped items from merge.

Commit:
Merge branch 'feature/fearow' of https://github.com/CenterForOpenScience/osf.io into COS-feature/fearow

Commit:
Merge branch 'COS-feature/fearow' into feature/validate-unreg-contributor-email

Commit:
Fixing permissions.READ for test

pattisdr committed:
Merge pull request #9046 from corbinSanders/feature/validate-unreg-contributor-email

Adding validation to APIv2 unregistered contributor email [ENG-56]

pattisdr committed:
Merge pull request #9035 from corbinSanders/feature/admin-userview-preprint

Feature/admin userview preprint [ENG-414]

brianjgeiger committed:
Add private node data collection
Also, Sydney is spelled like that, and not 'Syndey'

corbinSanders committed:
Merge pull request #6 from CenterForOpenScience/develop

Bringing feature/self-linking-validation up to date with osf.io/develop

Commit:
Fixing tests for new cyclic node linking check

pattisdr committed:
Merge branch 'develop' into feature/meetings-optimization

pattisdr committed:
Bump version and update changelog.

pattisdr committed:
Merge branch 'release/19.14.0'

pattisdr committed:
Merge branch 'release/19.14.0' into develop

brianjgeiger committed:
Change script runing start time

brianjgeiger committed:
Disable schedule until manually verified on Prod

Johnetordoff committed:
fix fangorn throttle issue by using queueable growl box over modal

pattisdr committed:
Merge pull request #9050 from brianjgeiger/feature/storage-metrics

[ENG-220] Data storage usage metrics

pattisdr committed:
Merge pull request #9077 from Johnetordoff/fangorn-throttle-fix

[ENG-142] Fix Fangorn rate throttling 202s

pattisdr committed:
Bump version and update changelog.

pattisdr committed:
Merge branch 'release/19.15.0'

pattisdr committed:
Merge branch 'release/19.15.0' into develop

UdayVarkhedkar committed:
Updates OSFUser social_links Property

* social_links will provide formatted social URLs for all social fields
* social_links will return a single url for social services that permit multiple accounts
* social_links will enforce typing dictated by SOCIAL_FIELDS

UdayVarkhedkar committed:
Updates OSFUser social_links Property

* Adds explanatory comment

Commit:
Adding ampersand sanitation to the components modal dialog

pattisdr committed:
Merge pull request #9082 from corbinSanders/feature/ampersand-display-fix

[ENG-53] Adding ampersand sanitation to the components modal dialog

Commit:
DRYing up the node linking error checks and fixing an error message.

Commit:
Modified method of creating the redirect to dashboard link

Commit:
Fixing order of parent/child node id's in error messages

pattisdr committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/editable-registrations

Johnetordoff committed:
make growlboxes blue for fangorn throttle issue

pattisdr committed:
Merge pull request #9083 from Johnetordoff/fix-fangorn-growlbox

[ENG-142] Make growlboxes blue for fangorn throttle issue

pattisdr committed:
Merge pull request #9085 from CenterForOpenScience/feature/raichu

Feature/Raichu - Fangorn Rate Limiting

pattisdr committed:
Bump version and update changelog.

pattisdr committed:
Merge branch 'release/19.16.0'

pattisdr committed:
Merge branch 'release/19.16.0' into develop

pattisdr committed:
Get up to date with osf.io/develop.

pattisdr committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/ENG505_schema_select_for_update

# Conflicts:
#	website/project/views/drafts.py

pattisdr committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/ENG141_update_as_predicted

pattisdr committed:
Rebase migration.

pattisdr committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/registration_files_count

# Conflicts:
#	api/registrations/serializers.py

pattisdr committed:
Rebase migration.

pattisdr committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/ENG463_registration_states_auto_approval

pattisdr committed:
Merge pull request #9013 from pattisdr/feature/ENG120_notification_subscriptions

[QA "Migration Prep" Required] Feature/Remove NotificationSubscriptions from Registrations [ENG-120]

pattisdr committed:
Merge pull request #9039 from pattisdr/feature/ENG505_schema_select_for_update

Feature/RegistrationSchema select_for_update [ENG-505]

pattisdr committed:
Merge pull request #9029 from pattisdr/feature/ENG463_registration_states_auto_approval

[QA Prep Required] Feature/Fix EmbargoTerminationApproval Status on Auto-Approve [ENG-463]

pattisdr committed:
Merge pull request #9015 from pattisdr/feature/ENG141_update_as_predicted

[QA Prep Required][Migration Required] Feature/Update AsPredicted form to the latest version [ENG-141]

pattisdr committed:
Merge branch 'feature/sandshrew' of https://github.com/CenterForOpenScience/osf.io into feature/registration_files_count

pattisdr committed:
Rebase migration.

pattisdr committed:
Merge pull request #9010 from pattisdr/feature/registration_files_count

[QA Prep Required][Migration Prep Required] Feature/Registrations Files Count [ENG-165]

pattisdr committed:
Remove progressbar in favor of tqdm.

pattisdr committed:
Merge pull request #9086 from CenterForOpenScience/feature/fearow

Merge of feature/fearow into develop

pattisdr committed:
Bump version and update changelog.

pattisdr committed:
Merge branch 'release/19.17.0'

pattisdr committed:
Merge branch 'release/19.17.0' into develop

pattisdr committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/sandshrew

Commit:
Add migration for ab_testing_home_page_version_b waffle flag

pattisdr committed:
Merge pull request #9065 from adlius/add-collected-metadata-relationship

[ENG-261] Add collected-metadata relationship to collection serializer.

adlius committed:
send license info to chronos

adlius committed:
Change error message

pattisdr committed:
Update aspredicted schema to more closely match official schema.

- Remove colon on register_draft template after each question.  Rely on original question punctuation.

pattisdr committed:
Merge pull request #9090 from pattisdr/feature/ENG669_as_predicted_improvements

[ENG-669] Feature/Update AsPredicted schema

Johnetordoff committed:
add storage usage calculations to new v2 WB hooks

pattisdr committed:
Fix duplicate qid on aspredicted template.

pattisdr committed:
Edit templates to let users know that their registrations will be made public automatically after some time period if they don't approve/cancel the embargo termination request themselves.

UdayVarkhedkar committed:
Updates social link tests to reflect storage of twitter/github as arrays

UdayVarkhedkar committed:
Adds more social links to pytest

pattisdr committed:
Merge pull request #9092 from pattisdr/feature/ENG669_more_as_predicted_improvements

Feature/More AsPredicted Improvements [ENG-669]

pattisdr committed:
Merge pull request #9074 from brianjgeiger/feature/visible-impact

[ENG-436] Fix preprint metrics

UdayVarkhedkar committed:
Imports past basestring for py2/3 compatibility

adlius committed:
use sync_manuscript insteand

pattisdr committed:
Merge pull request #9096 from CenterForOpenScience/feature/sandshrew

Merge Feature/sandshrew

pattisdr committed:
Bump version and update changelog.

pattisdr committed:
Merge branch 'release/19.18.0'

pattisdr committed:
Merge branch 'release/19.18.0' into develop

pattisdr committed:
Merge branch 'develop' into feature/PLAT1771_pagecounter_migrate. Rebase migrations.

Commit:
Removed the Auth assignment from retract_registrations, added in a log message for retract_registration_no_user

Johnetordoff committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into crossref-set-old-DOIs

* 'develop' of https://github.com/CenterForOpenScience/osf.io: (331 commits)
  Bump version and update changelog.
  Edit templates to let users know that their registrations will be made public automatically after some time period if they don't approve/cancel the embargo termination request themselves.
  Fix duplicate qid on aspredicted template.
  Update aspredicted schema to more closely match official schema.
  Bump version and update changelog.
  Remove progressbar in favor of tqdm.
  Rebase migration.
  Rebase migration.
  Rebase migration.
  Bump version and update changelog.
  make growlboxes blue for fangorn throttle issue
  Adding ampersand sanitation to the components modal dialog
  Bump version and update changelog.
  fix fangorn throttle issue by using queueable growl box over modal
  Disable schedule until manually verified on Prod
  Change script runing start time
  Bump version and update changelog.
  Add private node data collection Also, Sydney is spelled like that, and not 'Syndey'
  Fixing permissions.READ for test
  Clean up, plus aggressive api versioning
  ...

Johnetordoff committed:
clean up logging statement for DOIs

Johnetordoff committed:
save DOI identifier values

Johnetordoff committed:
spruce up emails for stuck crossref DOIs

adlius committed:
fix test

Johnetordoff committed:
make dry_run default to true for DOIs

Johnetordoff committed:
change crossref pending period and fix type

Johnetordoff committed:
exclude preprints that have only been "stuck" a short time from unsticking process

Johnetordoff committed:
make django-waffle work with flask and ember

Johnetordoff committed:
fix/clean-up crossref DOI checking

Johnetordoff committed:
add check for DOI to CeleryConfigs

Johnetordoff committed:
gotta pin to trusty for now, my default build xenial is bugged with ES

Johnetordoff committed:
move tests into TCI/linting range

UdayVarkhedkar committed:
Modifies Google Drive NodeLog message to exclude path of folder

UdayVarkhedkar committed:
Merge branch 'develop' into feature/fix-google-drive-path-logging

pattisdr committed:
Move article_doi and set_article_doi to AbstractNode model.

pattisdr committed:
Merge branch 'feature/editable-registrations' of https://github.com/CenterForOpenScience/osf.io into feature/ENG557_editable_registration_logging

pattisdr committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/PLAT1062_apitokenscope

pattisdr committed:
Rebase token/scopes migrations, define scope version as a variable, and bump API version.

Johnetordoff committed:
try more stuff for travis problems

Johnetordoff committed:
try more stuff for travis problems

brianjgeiger committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/registration_metrics

# Conflicts:
#	website/settings/defaults.py

brianjgeiger committed:
DRY up row encoding

pattisdr committed:
Merge pull request #9064 from Johnetordoff/crossref-set-old-DOIs

[ENG-373] Create periodic task that checks and sets stuck DOIs

pattisdr committed:
Merge pull request #9075 from brianjgeiger/feature/registration_metrics

[ENG-633] Add registration metrics gathering script

Commit:
Improving language for registration logs

Johnetordoff committed:
make task check DOIs in batches of 20

pattisdr committed:
Merge pull request #9094 from adlius/ENG-650_sync_chronos_status

[ENG-650] Use sync_manuscript for ChronosSubmissionDetailSerializer

Johnetordoff committed:
move helpful comment

Johnetordoff committed:
remove wafflejs urls and comments

Johnetordoff committed:
move and fix waffle tests

Johnetordoff committed:
fix waffle test fixture

Commit:
Making modifications to tests

Johnetordoff committed:
move comment to docstring

pattisdr committed:
Merge pull request #9102 from Johnetordoff/feature/nidorino-crossref

Make DOI tasks check crossref in batches

Commit:
Rename migration and fix dependency

Commit:
Remove outdated migration

Johnetordoff committed:
fix for travis

Johnetordoff committed:
fix for travis

Johnetordoff committed:
remove out of context request checking

Johnetordoff committed:
fix dupe test method names for storage usage tests

adlius committed:
add license check

Johnetordoff committed:
fix intra-node copy tests for storage usage and make tests more readable

Johnetordoff committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into wb-hook-storage-usage

* 'develop' of https://github.com/CenterForOpenScience/osf.io: (32 commits)
  DRY up row encoding
  fix test
  Bump version and update changelog.
  use sync_manuscript insteand
  Edit templates to let users know that their registrations will be made public automatically after some time period if they don't approve/cancel the embargo termination request themselves.
  Fix duplicate qid on aspredicted template.
  Update aspredicted schema to more closely match official schema.
  Remove progressbar in favor of tqdm.
  Rebase migration.
  Rebase migration.
  Rebase migration.
  Add registration metrics gathering script
  Fewer changes are better changes (keep formatting the same as it was)
  Have metrics return totals from histograms rather than counts
  Enable future Impact indices to have source fields
  Add comment about intentional decision to import Registration model at the top of the page.
  Don't load RegistrationSchemas via get_or_http_error which calls select_for_update() which locks the row.
  Print out unique attached registration guids.
  Add test to enforce that cron job marks embargo_termination approvals as APPROVED.
  Fix cron job that automatically approves embargo terminations - marks EmbargoTerminationApproval itself as approved.
  ...

Johnetordoff committed:
fix for travis

UdayVarkhedkar committed:
Improve readability of social_links

Johnetordoff committed:
fix so tests dont have to use celery

pattisdr committed:
Merge pull request #9089 from adlius/ENG-656_send_license_to_Chronos

[ENG-655][ENG-656] Send license info to Chronos

Commit:
Merge branch 'develop' into feature/ENG_63_removing_collections_admin_app

pattisdr committed:
Merge pull request #9103 from CenterForOpenScience/feature/nidorino-crossref

Feature/Merge Nidorino

Commit:
Travis fix

Johnetordoff committed:
merge get-waffle-cookies with develop

Commit:
Adding pytest mark for bookmark creation

Johnetordoff committed:
try to fix for travis

pattisdr committed:
Bump version and update changelog.

pattisdr committed:
Merge branch 'release/19.19.0'

pattisdr committed:
Merge branch 'release/19.19.0' into develop

Johnetordoff committed:
move __future__ import to test files only

Johnetordoff committed:
remove exception catching for request context error

Johnetordoff committed:
more fixes for travis

Commit:
Fixing test

Johnetordoff committed:
remove thrown exception for travis

Johnetordoff committed:
getcwd for Travis

jamescdavis committed:
feat: expose preprint request actions to requester and preprint admins

Johnetordoff committed:
make osftestcase into pytest for waffle cookie

Johnetordoff committed:
try new travis config for waffle cookies

mfraezz committed:
Update FAQ link

pattisdr committed:
Merge pull request #9088 from baylee-d/feature/ENG-661-feature-flag-migration

[ENG-661] Add migration for ab_testing_home_page_version_b waffle flag

pattisdr committed:
Merge pull request #9084 from Johnetordoff/get-waffle-cookies

[ENG-643] Get Waffle Cookies working!

jamescdavis committed:
Merge pull request #9104 from jamescdavis/ENG-736_expose_preprint_request_actions_to_preprint_admins

[ENG-736] expose preprint request actions to Preprint admins

mfraezz committed:
Merge branch 'hotfix/19.19.1'

mfraezz committed:
Merge branch 'hotfix/19.19.1' into develop

Johnetordoff committed:
isolate bug

Johnetordoff committed:
leave Travis confused, but completely unharmed

pattisdr committed:
Merge pull request #9107 from Johnetordoff/get-waffle-cookies

Get waffle cookies working... again!

pattisdr committed:
Merge pull request #9109 from CenterForOpenScience/feature/waffle_cookies

Merge of Feature/waffle cookies into develop

pattisdr committed:
Merge pull request #9110 from CenterForOpenScience/feature/preprint_request_actions_perms_easing

Merge "preprint request actions perms easing"  into develop

pattisdr committed:
Bump version and update changelog.

pattisdr committed:
Merge branch 'release/19.20.0'

pattisdr committed:
Merge branch 'release/19.20.0' into develop

pattisdr committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/PLAT1771_pagecounter_migrate

pattisdr committed:
Rebase migrations.

pattisdr committed:
Merge pull request #8800 from pattisdr/feature/PLAT1771_pagecounter_migrate

Feature/Optimize PageCounter Queries Release One [PLAT-1171]

brianjgeiger committed:
Update metrics scripts
1. Ensure the script has access to remote base folder regardless of where it's called from
2. Import the management commands to celery
3. Enable data storage usage command by default
4. Disable running keen snapshots by default (because they are slow)

brianjgeiger committed:
Re-disable data_storage_usage by default

So we don't break stagings

mfraezz committed:
Merge branch 'hotfix/19.20.1'

mfraezz committed:
Merge branch 'hotfix/19.20.1' into develop

mfraezz committed:
Fix import paths

mfraezz committed:
Merge branch 'hotfix/19.20.2'

mfraezz committed:
Merge branch 'hotfix/19.20.2' into develop

cslzchen committed:
Revert a waffle change that causes institution auth to break

- The commit that introduces the change: https://github.com/CenterForOpenScience/osf.io/commit/6e65c870bd67b8f4d034706f0cea3d6243643f86
- The PR that includes the above commit: https://github.com/CenterForOpenScience/osf.io/pull/9084

cslzchen committed:
Add comments for InstitutionAuth and its authentication class

Briefly explain what the endpoint does. Add a warning to check
with code owners before making future changes.

mfraezz committed:
Merge branch 'hotfix/19.20.3'

mfraezz committed:
Merge branch 'hotfix/19.20.3' into develop

pattisdr committed:
When calling get_or_create on existing PageCounter record, don't additionally query on the new resource, file, action, and version fields until all rows are populated with the management command.  (This will come with part II of this work).

pattisdr committed:
Merge pull request #9117 from pattisdr/feature/page-counter-part-1

[Needs Migration Prep] Feature/PageCounter Part I Fix [ENG-122][ENG-751][ENG-752]

aaxelb committed:
Merge pull request #9067 from pattisdr/feature/ENG557_editable_registration_logging

Feature/Editable Category and Logging[ENG-557]

aaxelb committed:
Merge branch 'develop' into feature/editable-registrations

pattisdr committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/spearow-page-counters

pattisdr committed:
Turn migrate_pagecounters into an unscheduled beat job.

pattisdr committed:
Merge pull request #9121 from pattisdr/feature/add_beat_job_for_pagecounters

Feature/Beat Job for Page Counters[ENG-122]

UdayVarkhedkar committed:
Merge branch 'develop' into feature/fix-broken-social-links

pattisdr committed:
Merge pull request #9125 from CenterForOpenScience/feature/spearow-page-counters

Feature/Merge spearow-page-counters branch

pattisdr committed:
Bump version and update changelog.

pattisdr committed:
Merge branch 'release/19.21.0'

pattisdr committed:
Merge branch 'release/19.21.0' into develop

pattisdr committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/editable-registrations

cslzchen committed:
Refactor the institution popuating script

Use `argparse` to specify the server environment to run the script on
and to select the institution(s) to update or create. Fixed style and
improved coding in general. Added new environment for staging 3.

cslzchen committed:
Update description for OU (The University of Oklahoma)

pattisdr committed:
Bump version that starts serializing subjects as relationships instead of attributes to 2.16.   (Pre 2.16, subjects still serialized as attributes).

- Reduce hardcoding of 2.16.

pattisdr committed:
Merge pull request #9129 from pattisdr/feature/ENG161_followup_editable_subjects

Feature/Bump Version to Serialize Subjects as Relationships [ENG-161]

cslzchen committed:
Log error and exit 1 when invalid IDs are provided via -d --ids

pattisdr committed:
Merge pull request #9128 from CenterForOpenScience/feature/editable-registrations

Feature/Merge of Editable Registrations into Develop

cslzchen committed:
Add an option to run the inst script main() with default args

This fixes the test failure when `main()` is imported and run as a
module in `osf_tests/test_elastic_search.py::TestSearchMigration`
by setting `default_args=True` during setup. Running the script via
command line is not affected using `default_args=False`.

cslzchen committed:
Move init_app() out of main() and let it run right before main()

pattisdr committed:
Bump version and update changelog.

pattisdr committed:
Merge branch 'release/19.22.0'

pattisdr committed:
Merge branch 'release/19.22.0' into develop

pattisdr committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/PLAT1062_apitokenscope

pattisdr committed:
Rebase migrations and bump API version.

pattisdr committed:
Merge pull request #8766 from pattisdr/feature/PLAT1062_apitokenscope

[QA Migration Prep Required] Feature/Token-Scope relationships [PLAT-1062]

pattisdr committed:
Bump version and update changelog.

pattisdr committed:
Merge branch 'release/19.23.0'

pattisdr committed:
Merge branch 'release/19.23.0' into develop

mfraezz committed:
Merge #9127

mfraezz committed:
Merge branch 'hotfix/19.23.1'

mfraezz committed:
Merge branch 'hotfix/19.23.1' into develop

UdayVarkhedkar committed:
Merge branch 'develop' into feature/fix-broken-social-links

UdayVarkhedkar committed:
Merge branch 'develop' into feature/fix-pending-embargoed-registration-types

UdayVarkhedkar committed:
Merge branch 'develop' into feature/fix-google-drive-path-logging

UdayVarkhedkar committed:
Merge branch 'develop' into feature/update_password_reset

pattisdr committed:
Merge pull request #9072 from UdayVarkhedkar/feature/fix-pending-embargoed-registration-types

[ENG-349] Feature/fix pending embargoed registration types

pattisdr committed:
Merge pull request #9071 from corbinSanders/feature/ENG_49_forgot_password_redirect

Reset Password page breaks login page[ENG-49]

pattisdr committed:
Merge pull request #9100 from UdayVarkhedkar/feature/fix-google-drive-path-logging

Feature/fix google drive path logging [ENG-692]

cslzchen committed:
Optimize the osf4i image (56.2%)

cslzchen committed:
New institution: University of North Carolina at Chapel Hill (UNC)

pattisdr committed:
Merge branch 'develop' into feature/api-post-v2-nodes-license

# Conflicts:
#	api/nodes/serializers.py

mfraezz committed:
Merge branch 'hotfix/19.23.2'

mfraezz committed:
Merge branch 'hotfix/19.23.2' into develop

pattisdr committed:
Merge branch 'develop' into wb-hook-storage-usage

pattisdr committed:
Remove extra trusty pinning.

pattisdr committed:
Merge pull request #9044 from corbinSanders/feature/api-post-v2-nodes-license

Add node license creation capability to APIv2 [ENG-51]

pattisdr committed:
Merge pull request #9051 from corbinSanders/feature/self-linking-validation

Linking a node to itself or a child [ENG-59]

pattisdr committed:
Merge branch 'develop' into feature/ENG_63_removing_collections_admin_app

pattisdr committed:
Remove extra trusty pinning.

pattisdr committed:
Merge pull request #9076 from corbinSanders/feature/ENG_63_removing_collections_admin_app

[ENG-63] Removing collections from user view in the django admin app

pattisdr committed:
Merge pull request #9091 from Johnetordoff/wb-hook-storage-usage

[ENG-356] Update Storage Usage for WB APIv2 move/copy hooks

pattisdr committed:
Merge pull request #9066 from UdayVarkhedkar/feature/update_password_reset

[ENG-60] Feature/update password reset

pattisdr committed:
Merge pull request #9079 from UdayVarkhedkar/feature/fix-broken-social-links

[ENG-552] Fix Broken Social Links

pattisdr committed:
Merge pull request #9097 from corbinSanders/feature/withdrawal_cron_fix

Log fix for automatic registration withdrawal after 48 hours [ENG-359]

cslzchen committed:
Add Universal Sustainability Hub (USH) to OSF Institutions

pattisdr committed:
Merge pull request #8765 from erinspace/feature/recursive_wb_move_copy_hooks

Recursively update most recent fileversion for folder contents [PLAT-1118]

cslzchen committed:
Add University of British Columbia (UBC) [Test] to OSF Institutions

pattisdr committed:
Bump version and update changelog.

pattisdr committed:
Merge branch 'release/19.24.0'

pattisdr committed:
Merge branch 'release/19.24.0' into develop

mfraezz committed:
Merge #9142

mfraezz committed:
Merge #9143

mfraezz committed:
Merge branch 'hotfix/19.24.1' into develop

pattisdr committed:
Merge branch 'develop' into feature/PLAT1771_pagecounter_migrate_Part_Two

# Conflicts:
#	osf/models/analytics.py
#	website/conferences/views.py

pattisdr committed:
Now that new pagecounter fields are populated, assert that action, file, and resource fields can't be None.

pattisdr commented:
closing and basing against COS/osf.io.
